packages/next-swc/crates/next-core/src/next_shared/transforms/postcss.rs

```diff
@@ -0,0 +1,112 @@
+use anyhow::Result;
+use turbo_tasks::{RcStr, Vc};
+use turbo_tasks_fs::FileSystemPath;
+use turbopack_binding::turbopack::{
+    core::{
+        resolve::options::{ImportMap, ImportMapping},
+        source::Source,
+        source_transform::SourceTransform,
+    },
+    node::{
+        execution_context::ExecutionContext,
+        transforms::postcss::{PostCssTransform, PostCssTransformOptions},
+    },
+    turbopack::{
+        evaluate_context::node_evaluate_asset_context,
+        module_options::{ModuleRule, ModuleRuleCondition, ModuleRuleEffect},
+    },
+};
+
+#[turbo_tasks::value]
+pub struct PostCssWrapperTransform {
+    execution_context: Vc<ExecutionContext>,
+    transform_options: Vc<PostCssTransformOptions>,
+}
+
+#[turbo_tasks::value_impl]
+impl PostCssWrapperTransform {
+    #[turbo_tasks::function]
+    pub fn new(
+        execution_context: Vc<ExecutionContext>,
+        transform_options: Vc<PostCssTransformOptions>,
+    ) -> Vc<Self> {
+        PostCssWrapperTransform {
+            execution_context,
+            transform_options,
+        }
+        .cell()
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl SourceTransform for PostCssWrapperTransform {
+    #[turbo_tasks::function]
+    async fn transform(self: Vc<Self>, source: Vc<Box<dyn Source>>) -> Result<Vc<Box<dyn Source>>> {
+        let this = self.await?;
+        let transform_options = this.transform_options.await?;
+
+        let import_map = if let Some(postcss_package) = transform_options.postcss_package {
+            package_import_map_from_import_mapping("postcss".into(), postcss_package)
+        } else {
+            let path = source.ident().path();
+            package_import_map_from_context("postcss".into(), path)
+        };
+
+        let evaluate_context = node_evaluate_asset_context(
+            this.execution_context,
+            Some(import_map),
+            None,
+            "postcss".into(),
+        );
+
+        Ok(PostCssTransform::new(
```

Comment:
@sokra what do you think about moving all postcss code (there's still some left in `turbopack-node` on the turbo side) into this crate, rather than wrapping that transform here?

Reply:
I think PostCSS handling should stay in turbo since it's not next.js specific and other frameworks or standalone Turbopack probably also want to use that.